src/main/scala/com/microsoft/hyperspace/index/rules/RuleUtils.scala

```diff
@@ -112,8 +112,16 @@ object RuleUtils {
               _,
               _,
               _) =>
-            location.allFiles.map(f =>
-              FileInfo(f.getPath.toString, f.getLen, f.getModificationTime))
+            location.allFiles.map(
+              f =>
+                // For a given file, file id is only meaningful in the context of a given
+                // index. At this point, we do not know which index, if any, would be picked.
+                // Therefore, we simply set the file id to UNKNOWN_FILE_ID.
+                FileInfo(
```

Comment:
Not related to this PR, but question: @sezruby can we not use `FileInfo` here (so that we can remove `UNKNOWN_FILE_ID`)?

Reply:
@imback82 `FileInfo` now has `id` and `id` is meaningful in the context of a given index. At this point, we dont really know which index is gonna be picked, if any. So `id` is really unknown here.

Reply:
Can't `isHybridScanCandidate` updated to take in `Seq[FileStatus]` instead? Seems like we need to change only `val commonCnt = inputSourceFiles.count(entry.sourceFileInfoSet.contains)` to work with `FileStatus`? (btw, we don't need to address this in this PR)

Reply:
Yes it's not necessarily FileInfo and also we can utilize tag feature to keep appended and deleted file info in getCandidateIndexes. I'll create an issue for this. Thanks

Reply:
Thanks!